internal/xds/translator/listener.go

```diff
@@ -293,18 +293,32 @@ func (t *Translator) addXdsHTTPFilterChain(xdsListener *listenerv3.Listener, irL
 		}
 
 		xdsListener.FilterChains = append(xdsListener.FilterChains, filterChain)
-	} else {
-		// Add the HTTP filter chain as the default filter chain
-		// Make sure one does not exist
-		if xdsListener.DefaultFilterChain != nil {
-			return errors.New("default filter chain already exists")
+	} else if !havePortMatch(xdsListener, irListener.Port) {
+		// No TLS, so create a matcher based on the port if one doesn't yet exist.
+		// Conflicts between listeners should have been detected in the GWAPI->IR
+		// translation level already.
+		filterChain.FilterChainMatch = &listenerv3.FilterChainMatch{
+			DestinationPort: wrapperspb.UInt32(irListener.Port),
 		}
-		xdsListener.DefaultFilterChain = filterChain
```

Comment:
raised in yesterday's community meeting: when matching on the destination port, wouldn't there be a dedicated envoy listener that's already bound to that distinct port? meaning, for port-based matching, the default filter chain can still be used and there's no need to additionally do a filter chain match?